src/ic-cdk-bindgen/CHANGELOG.md

```diff
@@ -6,6 +6,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [unreleased]
 
+- Breaking change: Look up the `CANISTER_ID_<canister_name_uppercase>` and
```

Comment:
IIUC, this doesn't have to be a breaking change.
Before this PR, `ic-cdk-codegen` works with existing `dfx` but may stop working with future `dfx` which removes the legacy env-var.
After this PR, `ic-cdk-codegen` should work with existing and future `dfx` versions.
From a user perspective, the fix doesn't break.

Reply:
That's true.  The breaking behavior I was thinking of would be for someone building with the CDK but not using dfx, if they didn't set both forms of environment variable.

Reply:
@lwshang PTAL

Reply:
I would like to make this PR non-breaking in any sense. That we make `ic-cdk-bindgen` look for the uppercase env vars firs, if not exist then look for the legacy ones .

And simply uppercasing the canister name is not accurate. We also need to replace hyphens with underscores.

I'd like to take over the following work. WDYT?